src/typescript/frontend/src/lib/hooks/queries/use-num-markets.ts

```diff
@@ -0,0 +1,20 @@
+import { RegistryView } from "@sdk/emojicoin_dot_fun/emojicoin-dot-fun";
+import { useQuery } from "@tanstack/react-query";
+import { getAptos } from "lib/utils/aptos-client";
+
+async function getNumMarkets(): Promise<number> {
+  const aptos = getAptos();
+  return RegistryView.view({ aptos }).then((res) => Number(res.n_markets));
+}
+
+export const useNumMarkets = () => {
+  const res = useQuery({
+    queryKey: ["num-markets"],
+    queryFn: () => {
+      return getNumMarkets();
+    },
+    staleTime: 30000,
```

Comment:
I think you can make this shorter, like 10-15 seconds, since it's from the REST API and it can handle it

You could also add the length of the market `symbols` map in `useEventStore` in the `queryKey`, which would trigger a re-fetch whenever it changes. You might be able to use a more appropriate way to do this with `tanstack/react-query`, but I think this would work:

```typescript
export const useNumMarkets = () => {
  const numMarkets = useEventStore((s) => s.symbols.size); // or whatever it is, I think it's a set so it's `size`
  // ...
    queryKey: ["num-markets", numMarkets],
  // ...
}
```

I believe that would work by basically triggering a re-fetch whenever the event store sees a new market come in. Then you get the best of both worlds, the node API + the live registrations coming in

Reply:
Thank you. I tried doing this with `tanstack/react-query` but I couldn't make it work, I'll add this !